bin/verify-attestation/src/main.rs

```diff
@@ -128,7 +128,27 @@ fn print_quote_verification_summary(quote_verification_result: &QuoteVerificatio
         println!("\tInfo: Advisory ID: {advisory}");
     }
     println!("Quote verification result: {}", tcblevel);
-    println!("mrsigner: {}", hex::encode(quote.report_body.mrsigner));
-    println!("mrenclave: {}", hex::encode(quote.report_body.mrenclave));
-    println!("reportdata: {}", hex::encode(quote.report_body.reportdata));
+
+    match &quote.report {
+        Report::SgxEnclave(report_body) => {
+            println!("mrsigner: {}", hex::encode(report_body.mr_signer));
+            println!("mrenclave: {}", hex::encode(report_body.mr_enclave));
+        }
+        Report::TD10(report_body) => {
+            println!("mrtd: {}", hex::encode(report_body.mr_td));
+            println!("rtmr0: {}", hex::encode(report_body.rt_mr0));
+            println!("rtmr1: {}", hex::encode(report_body.rt_mr1));
+            println!("rtmr2: {}", hex::encode(report_body.rt_mr2));
+            println!("rtmr3: {}", hex::encode(report_body.rt_mr3));
+        }
+        Report::TD15(report_body) => {
+            let report_body = &report_body.base;
+            println!("mrtd: {}", hex::encode(report_body.mr_td));
+            println!("rtmr0: {}", hex::encode(report_body.rt_mr0));
+            println!("rtmr1: {}", hex::encode(report_body.rt_mr1));
+            println!("rtmr2: {}", hex::encode(report_body.rt_mr2));
+            println!("rtmr3: {}", hex::encode(report_body.rt_mr3));
+        }
+    }
+    println!("reportdata: {}", hex::encode(quote.get_report_data()));
 }
```

Comment:
I don’t want to be too nit-picky here since you’re focused on researching and figuring out TDX, but there’s quite a bit of overlapping code between `bin/verify-era-proof-attestation/src/verification.rs` and `bin/verify-attestation/src/main.rs`. Maybe it would make sense to unify those two tools? (not necessarily in this PR)

Reply:
maybe something can be factored out... `verify-attestation` was a test tool... `verify-era-proof-attestation` is in production and specialized

Reply:
Maybe it’s worth implementing the `fmt::Display` trait for `dcap_qvl::quote::Quote` or `dcap_qvl::quote::Report`, similar to how you did with the `GetReportData` trait. This way, you could reuse it across both binaries. Just nitpicking – feel free to ignore for now.

Reply:
implemented.. including the alternate version with newlines